src/compile/data/optimize.ts

```diff
@@ -47,35 +47,34 @@ function runOptimizer(optimizer: BottomUpOptimizer | TopDownOptimizer, nodes: Da
       return optimizer.run(node);
     }
   });
-  return flags.some(isTrue) || flag;
+  return flags.some(isTrue);
 }
 
 function optimizationDataflowHelper(dataComponent: DataComponent, model: Model) {
   let roots = dataComponent.sources;
-  let mutatedFlag = false;
+  const mutatedFlags: boolean[] = [];
```

Comment:
What if we make this a `Set` and then use `add`? Then we only need to have at most two items in the set.

Reply:
Yeah, that's probably better. 
The return value would then be `<setName>.has(true)`.

Reply:
Let me know when you had a chance to finish this.

Reply:
Pushed changes :)